Add introduction to product focused learning path.

MaineC opened the pull request:
This adds the introduction to the product focused learning path.

lenucksi reviewed (commented):
That's a nice write-up. 
It reads a bit more like "combining Agile and InnerSource / where's the difference and what do I do as a team lead". 

Wondering where this fits in though and what's the planned future?
- #62 ? #274? #488?

We tried the outline & scope first with the other parts - do you think that could be a worthwhile approach here too? 
(If nothing else: Good content, yay! :tada: )

MaineC commented:
As for the focus: The goal is to answer all questions related to InnerSource some product manager might have. My challenge is that my own product management experience is limited. Thus so far this mirrors the questions I have gotten. Happy to include other aspects people have received as well.

As for other issues:

* My personal plan is to get a text version done until very early 2023.
* I would love to see videos, but that depends on finding a co-speaker (I find the learning path segments more lively if there are two faces) and on finding sponsoring to do any shooting.

Outline: Have you seen https://github.com/InnerSourceCommons/InnerSourceLearningPath/blob/main/product/outline.md ?

lenucksi commented:
> As for other issues:
> 
> * My personal plan is to get a text version done until very early 2023.
> * I would love to see videos, but that depends on finding a co-speaker (I find the learning path segments more lively if there are two faces) and on finding sponsoring to do any shooting.

Got it.
> 
> Outline: Have you seen https://github.com/InnerSourceCommons/InnerSourceLearningPath/blob/main/product/outline.md ?

 Yes, too long ago. I forgot the connection. Now the connection is there, though. :)

rrrutledge reviewed (commented):
Thank you for this work, @MaineC ❗   We are very glad for this effort and teaching that you are putting in to this new section of the Learning Path.

Can you kindly put each sentence on a new line in the markdown source?  The text will still render the same in _GitHub_, but this practice makes it easier to leave review comments on a particular sentence.  See [Contributing.md](https://github.com/InnerSourceCommons/InnerSourceLearningPath/blob/main/CONTRIBUTING.md#files).  Thank you!

MaineC commented:
I've changed the formatting. 

I also added the content for the "relation between agile and InnerSource" segment, however so far I tried to stick with outline-like bullet points only. Thanks for that reminder @lenucksi

rrrutledge commented on product/02-innersource-and-agile.md:
```diff
@@ -0,0 +1,82 @@
+That means you are losing some level of control.
+You are also losing direct oversight. As a result performance feedback for InnerSource contributors needs to take the perspective of the host team into account.
+
+* Impact of language: InnerSource and Agile uses some of the same tooling - for different purposes.
```
`Product Owner` in InnerSource needs to negotiate and align on off-team contributions (not just work for the host team).

MaineC replied:
I've tried to address that in the negotiation part.

rrrutledge replied:
OK maybe just at least a mention of it, then?  It sounds like a common misunderstanding and so seems appropriate to mention here even if it is explained in detail somewhere else.

```suggestion
* Impact of language: InnerSource and Agile uses some of the same tooling - for different purposes.
   - Product Owner needs to support and align on off-team contributions (rather than strictly work with the Host team).
More to come on this in **InnerSource and Negotiation**.
```

rrrutledge commented on product/02-innersource-and-agile.md:
```diff
@@ -0,0 +1,82 @@
+   - Existing test driven development means that confidence in contributions is higher.
+   - XP teams have a "leave the code in better shape than you found it" attitude - mitigates the risk of decline in quality or co-hesion of code when faced with multiple contributions from different sources.
+
+## Common misunderstandings when coming from agile teams
```
Separation of ownership for a feature - Host team ultimately owns the feature design and ongoing maintenance.  Guest team owns the code creation.

MaineC replied:
I've tried to address that in the fifth section below.

rrrutledge commented on product/02-innersource-and-agile.md:
```diff
@@ -0,0 +1,82 @@
+In InnerSource for a conversation with the customer, but also for communication between members of a team of trusted committers and contributors working on one common InnerSource component.
+Issues in InnerSource become much more lengthy and wordy than in your average organisation.
+
+  - Code reviews: In traditional organisations those often serve auditing purposes and for quality controle.
```
```suggestion
  - Code reviews: In traditional organisations those often serve auditing purposes and for quality control.
```

rrrutledge reviewed (changes requested):
Thank you, @MaineC ❗   We are very excited for this work!

MaineC commented on product/02-innersource-and-agile.md:
```diff
@@ -0,0 +1,82 @@
+
+## Common misunderstandings when coming from agile teams
+
+* Impact on performance reviews: InnerSource moves some of the energy you spend developing outside of your own team.
```
Re-reading that bullet point now, it's in the wrong section.

In Germany at least companies have regular, often once a year, or quarterly or something like that, performance reviews where your line manager looks at what you accomplished and what your focus area should be in the coming months. Typically that manager is fairly close to your daily work so it's easy to talk about the accomplishments of previous weeks and months. As employees move to roles that have impact beyond their local teams that conversation needs to adjust. InnerSource is a bit special here: Contributors and Trusted Committers by their job title come from a "software engineer" background, but their impact may go across several hierarchical branches inside of the organisation - though a bit less than in Open Source: If in Open Source you work on backstage, foundation level topics, your work may impact even other corporations, but still be invisible - because private - to the person who agreed to donate your time to that foundation. In InnerSource all that remains within one corporation, but might still be out of sight for where your manager and your team typically operates.

MaineC replied:
Ultimately there need to be factors that go into KPIs. 

Contributing outside of your team and mentoring contributors is something that I would expect to become part of seniority levels.

If we look back to the "enlightened self interest" for open source contributions I could see something similar for InnerSource as well: 
* Move faster by standing on the shoulders of giants, combined with
* accomplish more by collaborating across business units (like we do here for the Learning Path segment - we all need it, it's not a unique selling point for neither of our employers, but by collaborating we can create something that would be more costly when done internally only), combined with
* fix issues that block shipping features to your customers and move maintenance to a team that is larger than your local team
In advanced settings I would expect people to understand the value of contributors working on simpler features freeing the host team to work on more complex changes that the contributors have a business need for.

MaineC replied:
Essentially I would like to capture some of the learnings that Myrle Krantz shared in her keynote at the InnerSource summit in that segment.

rrrutledge replied:
Is there any update to make to the outline to incorporate these ideas?

MaineC replied:
Good point - I'll see where these points fit.

MaineC commented:
I've added outline-like skeletons for the remaining sections.

rrrutledge commented:
Excited for these changes, @MaineC ❗️ Will take a look soon.

MaineC commented:
Also - as an invitation to others involved with teaching InnerSource to people with less experience in open source project management - any additional feedback welcome. @claredillon I believe you had also been talking to some folks who might have valuable input here?

spier commented:
FYi we have this early pattern draft:
[Explaining InnerSource to Management by anchoring it to Agile / DevOps / Lean](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/concept-anchor.md#solutions)

You might not want to link to it from the Learning Path content yet, as it hasn't been vetted by any org until now.
However it could still provide some inspiration for the bits related to Agile.

/cc @mishari

MaineC commented:
@spier thank you for the pointer, I hadn't thought of the pattern but it is a good idea to link to it.

spier commented:
> @spier thank you for the pointer, I hadn't thought of the pattern but it is a good idea to link to it.

@MaineC I meant this more for inspiration, and less for for linking.
Or at least if you link to it, be aware that that link can change at any point as that pattern is still in it's initial stage.

Once we level up the patterns to "Structured" and publish them to the book, the links can be expected to be stable.

MaineC commented:
@spier thanks for the reminder about links being unstable as long as the pattern is marked as initial...

MaineC commented:
I have a question: For the past few days I had a bit of time to work on the full text version (instead of just outline/ bullet points) for this learning path segment. I wouldn't want to simply push that over the discussion here, but create a separate pull request for it. Does anyone have any objections to doing that?

rrrutledge commented:
> create a separate pull request for it

Which discussion are we talking about, here?

I have a note to come back to review this PR again hopefully to merge soon - it feel like we are getting pretty close or already at that point.

MaineC commented:
> I have a note to come back to review this PR again hopefully to merge soon - it feel like we are getting pretty close or 
> already at that point.

That is why I think it might be best to post the full text version I have been typing up the past few days in a separate pull request that builds on this one. That way we can merge the work done here soon and continue to work on the full text in a separate pull request. Does that make sense?

rrrutledge reviewed (approved):
Yes. Separate pull request for the full text sounds great.
